empress/__init__.py

```diff
@@ -145,19 +147,19 @@ def median(self) -> ReconciliationWrapper:
         Return one of the best ReconciliationWrapper that best represents the
         reconciliation graph. The function internally uses random and is not deterministic.
         """
-        postorder_parasite_tree, gene_tree_root, _ = diameter.reformat_tree(self.recon_input.parasite_dict, "pTop")
+        postorder_parasite_tree, parasite_tree_root, _ = diameter.reformat_tree(self.recon_input.parasite_dict, "pTop")
         postorder_host_tree, _, _ = diameter.reformat_tree(self.recon_input.host_dict, "hTop")
 
         # Compute the median reconciliation graph
         median_reconciliation, n_meds, roots_for_median = median.get_median_graph(
-            self.recongraph, postorder_parasite_tree, postorder_host_tree, gene_tree_root, self.roots)
+            self.recongraph, postorder_parasite_tree, postorder_host_tree, parasite_tree_root, self.roots)
 
         med_counts_dict = median.get_med_counts(median_reconciliation, roots_for_median)
 
         random_median = median.choose_random_median_wrapper(median_reconciliation, roots_for_median, med_counts_dict)
         median_root = _find_roots(random_median)[0]
         return ReconciliationWrapper(random_median, median_root, self.recon_input, self.dup_cost, self.trans_cost,
-                                     self.loss_cost, self.total_cost)
+                                     self.loss_cost, self.total_cost, self.event_scores)
 
     def cluster(self, n) -> List['ReconGraphWrapper']:
```

Comment:
It seems that the code implicitly recompute event scores when you cluster, which is not the expected behavior.

Reply:
You might consider adding tests that make sure the event freq stays the same after you cluster or median, but I don't want to take more of your time.